packages/flutter/lib/src/material/menu_bar_theme.dart

```diff
@@ -41,8 +41,11 @@ class MenuBarThemeData extends MenuThemeData {
   /// Creates a const set of properties used to configure [MenuTheme].
   const MenuBarThemeData({super.style});
 
-  /// Linearly interpolate between two text button themes.
+  /// Linearly interpolate between two [MenuBar] themes.
   static MenuBarThemeData? lerp(MenuBarThemeData? a, MenuBarThemeData? b, double t) {
+    if (identical(a, b)) {
+      return a;
+    }
```

Comment:
There's a slight change of functionality here. Previously this would always return a MenuBarThemeData and never null (although it has a nullable return type). Now, it will return null if a and b both are null. Not sure if this matters...

(This is also true for a lot of other lerp methods adjusted in this PR, I am gonna refrain from commenting on all of them :))

Reply:
(Unrelated to this PR:) It is driving me crazy how inconsistent the lerp methods are with regards to nullability. Some allow nulls as a arguments, some don't. Some allow null as a return value, some don't. I wonder if we should audit this and standardize on one thing (or maybe its already too late to do this).

Reply:
I've added my notes about the variation in lerp methods to the PR's description.

The slight change in MenuBarThemeData.lerp() was intentional because the code was rather new (so new it still referred to its themes as "text buttons") so it seemed like it would be safe to normalize it a little.

Similar changes were only applied to lerp methods that already special-cased `(a == null && b == null)`. For the others (the ones without this special case) the translation was:
```dart
    if (identical(a, b) && a != null) {
      return a;
    }
 ```

Reply:
Sounds good. Thanks for adding the stats to the PR description.